### ISSUE TITLE
Gate does not start (Java Keystore file not found) when using Halyard 1.19.0

### ISSUE THREAD

**Issue by iniinikoski** [spinnaker/spinnaker#4294]
### Issue Summary:

When trying to "hal deploy apply" with SAML enabled - Gate does not start (complaining about missing Java Keystore file):

````
Caused by: java.io.FileNotFoundException: class path resource [home/halyard/.hal/default/staging/dependencies/1745581636-keystore.jks] cannot be opened because it does not exist
````

The keystore file does exist on the setup - and the permissions are fine. The only change in between two "hal deploy apply"'s are the version of Halyard (1.18.0 vs. 1.19.0).

### Cloud Provider(s):

Any.

### Environment:

Spinnaker version: 1.13.1 - running in GKE
Halyard version: 1.19.0 (broken) - 1.18.0 (working)

### Feature Area (if this issue is UI/UX related, please tag `@spinnaker/ui-ux-team`):

Halyard + Gate

### Steps to Reproduce:

Should be repeatable like this:

1. install Spinnaker with Halyard - using Halyard version 1.18.0 - configuring also SAML
2. Spinnaker gets deployed correct (also Gate starts fine)
3. upgrade Halyard to version 1.19.0 - and run "hal deploy apply"
4. witness Gate not starting (see below where the change is)

### Additional Details:

For some reason, Halyard configured the JKS location differently between Halyard versions, like this (generated config from ~/.hal/default/staging/gate.yml):

in 1.18.0 (working version):

````
saml:
  enabled: true
  issuerId: https://spinnaker.redbullmediahouse.com
  metadataUrl: file:/home/halyard/.hal/default/staging/dependencies/362827086-metadata.xml
  keyStore: file:/home/halyard/.hal/default/staging/dependencies/1745581636-keystore.jks
  keyStorePassword: topSecretPassword
  keyStoreAliasName: saml
  redirectProtocol: https
  redirectHostname: spinnaker-api.company.com
  userAttributeMapping: {}
````

in 1.19.0 (broken one):

````
saml:
  enabled: true
  issuerId: https://spinnaker.redbullmediahouse.com
  metadataUrl: file:/home/halyard/.hal/default/staging/dependencies/362827086-metadata.xml
  keyStore: /home/halyard/.hal/default/staging/dependencies/1745581636-keystore.jks
  keyStorePassword: topSecretPassword
  keyStoreAliasName: saml
  redirectProtocol: https
  redirectHostname: spinnaker-api.company.com
  userAttributeMapping: {}
````

**Comment by iniinikoski** [comment]
Hmm, could it be this one: https://github.com/spinnaker/halyard/commit/4762885a57389eb500ed6afe7fd6af9f978362cf ?

**Comment by devorbitus** [comment]
We found a workaround before finding this issue, it's not great but does seem to work.
Add this to gate-local.yml in `.hal/default/profiles/`
```yaml
saml: 
  keyStore: file:/home/halyard/.hal/default/staging/dependencies/1745581636-keystore.jks
```

**Comment by iniinikoski** [comment]
The fix seems to be already in here: https://github.com/spinnaker/halyard/pull/1276

**Comment by chinaboard** [comment]
@iniinikoski 
1.19.1 same problem

**Comment by devorbitus** [comment]
The fix is slated for the 1.20 release

**Comment by dibyom** [comment]
Fix has been cherry-picked on to halyard release 1.19.2

**Comment by devorbitus** [comment]
oh thank goodness, we keep getting bitten by this. Really appreciate the turn-around! :)

**Comment by iniinikoski** [comment]
Verified: fix is included in 1.19.2. Huge thanks!
